doc/NextGen_ON_CONUS.md

```diff
@@ -0,0 +1,173 @@
+# NextGen on CONUS
+
+This documentation provides instructions on all neccessary steps and components to run NextGen jobs at CONUS scale. Considering the computations large scale, we focus only on running parallel jobs using MPI.
+
+* [Summary](#summary)
+* [Download the Codes](#doenload-the-codes)
+* [Setting Up the Environment](#setting-up-the-environment)
+* [Build the Executable](#build-the-executable)
+* [CONUS Hydrofabric](#CONUS-hydrofabric)
+* [Generate Partition For Parallel Computation](#generate-partition-for-parallel-computation)
+* [Prepare the Input Data](#prepare-the-input-data)
+* [Build the Realization Configurations](#build-the-realization-configurations)
+* [Run Computations with submodules](#run-computations-with-submodules)
+* [Run Computation with Topmodel](#run-computation-with-topmodel)
+* [Run Computation with Routing](#run-computation-with-routing)
+
+# Summary
+
+This is a tutorial-like documentation. We provide suficient details in the hope that by following this document step by step, you can run NextGen computations from simple to sophisticated realization that models simple test examples to realistic cases. Throughout this document, we assume a Linux operating system environment.
+
+# Download the Codes
+
+To download the `ngen` source code, run the following commands:
+
+`git clone https://github.com/NOAA-OWP/ngen.git`
+`cd ngen`
+
+Then we need all the submodule codes. So run the command below:
+
+`git submodule update --init --recursive`
+
+# Setting up the Environment
+
+For setting up the build and computation environment, we refer the users to our documentation chapter [DEPENDENCIES.md](DEPENDENCIES.md) for details. Basically, you will need to have access to C/C++ compiler, MPI, Boost, NetCDF, Cmake, SQLite3. Some of them may already be on your system. Otherwise, you have to install your own version. There are also some required software packages that come with `ngen` as submodules, such as `Udunits libraries`, `pybind11`, and `iso_c_fortran_bmi`. 
+
+You are most likely need to use Python. For that we recommend setting up a virtual environment. For details, see [PYTHON_ROUTING.md](PYTHON_ROUTING.md). After setting up the Python virtual environment and activating it, you may need install additional python modules depending what `ngen submodules` you want to run.
+
+# Build the Executable
+
+After setting up the environment variables, we need first build the necessay dynamically linked librares. Although `ngen` has capability for automated building of submodule libraries, we build them explicitly so that users have a better understanding. For simplicity, we display the content a script which we name it `build_libs`.
```

Comment:
```suggestion
After setting up the environment variables, we need to first build the necessary dynamically linked libraries. Although `ngen` has the capability for automated building of submodule libraries, we build them explicitly so that users have a better understanding. For simplicity, we display the content a script which we name it `build_libs`.
```